front/src/modules/simulationResult/helpers/utils.ts

```diff
@@ -0,0 +1,113 @@
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainSchedule/utils';
+
+export const filterSignalUpdatesByPosition = (
+  signals: ProjectPathTrainResult['signal_updates'],
+  cutPositionStart: number,
+  cutPositionEnd: number
+): ProjectPathTrainResult['signal_updates'] =>
+  signals
+    .map((signal) => {
+      if (signal.position_start < cutPositionStart && signal.position_end > cutPositionStart) {
+        const interpolationFactor =
+          (cutPositionStart - signal.position_start) /
+          (signal.position_end - signal.position_start);
+        return {
+          ...signal,
+          position_start: cutPositionStart,
+          time_start:
+            signal.time_start + (signal.time_end - signal.time_start) * interpolationFactor,
+        };
+      }
+      if (signal.position_start < cutPositionEnd && signal.position_end > cutPositionEnd) {
+        const interpolationFactor =
+          (signal.position_end - cutPositionEnd) / (signal.position_end - signal.position_start);
+        return {
+          ...signal,
+          position_end: cutPositionEnd,
+          time_end: signal.time_end - (signal.time_end - signal.time_start) * interpolationFactor,
+        };
+      }
+      return signal;
+    })
+    .filter(
+      (signal) => signal.position_start >= cutPositionStart && signal.position_end <= cutPositionEnd
+    );
+
+export const filterSpaceTimeCurvesByPosition = (
+  spaceTimeCurves: ProjectPathTrainResult['space_time_curves'],
+  cutPositionStart: number,
+  cutPositionEnd: number
+): ProjectPathTrainResult['space_time_curves'] =>
+  spaceTimeCurves.map((spaceTimeCurve) => {
+    let { positions, times } = { ...spaceTimeCurve };
+
+    // If the position where we want to cut the curve isn't already in the curve positions,
+    // We need to interpolate the corresponding time and it and the position the curve
+
+    if (positions.at(0)! <= cutPositionStart && !positions.includes(cutPositionStart)) {
+      const interpolatedTime = interpolateValue(
+        { positions, speeds: [], times },
+        cutPositionStart,
+        'times'
+      );
+
+      positions = [cutPositionStart, ...positions];
+      times = [interpolatedTime, ...times];
+    }
+
+    if (positions.at(-1)! >= cutPositionEnd && !positions.includes(cutPositionEnd)) {
+      const interpolatedTime = interpolateValue(
+        { positions, speeds: [], times },
+        cutPositionEnd,
+        'times'
+      );
+
+      positions = [...positions, cutPositionEnd];
+      times = [...times, interpolatedTime];
+    }
+
+    const filteredIndex = positions
+      .map((position, index) =>
+        position >= cutPositionStart && position <= cutPositionEnd ? index : -1
+      )
+      .filter((index) => index !== -1);
+
+    return {
+      positions: filteredIndex.map((index) => positions[index]),
+      times: filteredIndex.map((index) => times[index]),
+    };
```

Comment:
Do you think it's clearer like this ?
```suggestion
    const result = { positions: [] as number[], times: [] as number[] };
    for (let i = 0; i < positions.length; i++) {
      const position = positions[i];
      if (position >= cutPositionStart && position <= cutPositionEnd) {
        result.positions.push(position);
        result.times.push(times[i]);
      }
    }
    return result;
```

Reply:
Only one loop instead of 3, great suggestion thank you :) Done